### PR TITLE
[noop] pre-commit linting

### PR DESCRIPTION
**Pull request by skiptomyliu**
Standardizing our coding with pre-commit.

### CODE REVIEW

**Comment by achantavy** [comment]
Is there a way to add this linter to the `make test`?

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -0,0 +1,42 @@
+repos:
```
Comment (skiptomyliu) [inline .pre-commit-config.yaml]: Add pre-commit config

File: Makefile [hunk 2]
```diff
@@ -1,7 +1,7 @@
 test: test_lint test_unit test_integration
 
 test_lint:
-	flake8
+	pre-commit run --all-files --show-diff-on-failure
```
Comment (skiptomyliu) [inline Makefile]: Replace `flake8` with `pre-commit` which runs `flake8`

File: test-requirements.txt [hunk 3]
```diff
@@ -1,3 +1,3 @@
-flake8
+pre-commit
```
Comment (skiptomyliu) [inline test-requirements.txt]: add `pre-commit` as a test dependency.

**Comment by skiptomyliu** [comment]
Thanks -- just pushed another commit to add it as a new dependency.

**Comment by ecdavis** [comment]
Is the double-quote string fixer necessary? I'm not sure what the benefit is and it's actually introduced inconsistencies because we _have_ to double-quote strings which contain single-quote characters (or escape the single-quotes... or change them to double-quotes I guess).

**Comment by skiptomyliu** [comment]
@ecdavis - not necessary, I've eliminated the `double-quote-string-fixer`, and have re-run it.

**Comment by achantavy** [comment]
Woot looks like this passed our internal tests.  Please update https://github.com/lyft/cartography/blob/master/docs/testing.md with the new mention of pre-commit and any other changes you think make sense

**Pull request merged**